File: 03 Ransom Note/Ransom Note.js
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function (ransomNote, magazine) {
    let mp = new Map();
    for (let i = 0; i < magazine.length; i++) {
        let x = 1;
        if (mp.has(magazine[i]))
            x += mp.get(magazine[i])
        mp.set(magazine[i], x);
    }
    for (let i = 0; i < ransomNote.length; i++) {
        if (!mp.has(ransomNote[i]) || mp.get(ransomNote[i]) == 0)
            return false;
        else
            mp.set(ransomNote[i], mp.get(ransomNote[i]) - 1);
    }
    return true;
};

File: 03 Ransom Note/Ransom Note.cpp
class Solution
{
public:
    bool canConstruct(string ransomNote, string magazine)
    {
        map<char, int> mp;
        for (int i = 0; i < magazine.size(); i++)
            mp[magazine[i]]++;
        for (int i = 0; i < ransomNote.size(); i++)
        {
            if (mp[ransomNote[i]] == 0)
                return false;
            else
                mp[ransomNote[i]]--;
        }
        return true;
    }
};